pkg/streamingpromql/README_FUZZ_TESTING.md

```diff
@@ -0,0 +1,137 @@
+# Query Fuzz Testing
+
+This file contains notes related to the addition of Fuzz tests for the Mimir query engine.
+
+Fuzz tests have been added to the `engine_fuzz_test.go` file. 
+
+These will run as a normal unit test, or they can be invoked with the Go Fuzz engine with `go test -fuzz=FuzzQuery`
+
+In either case, the test function loads a set of sample data from `seed-data.test` and then runs a range of sample queries from `seed-queries.test`.
+
+The test assertions are that the Prometheus PromQL Engine returns the same result as the Mimir Query Engine. Considered in this test are the result errors, result values and result annotations.
+
+There are some caveats to this comparison;
+
+* there are cases where query result annotations (error/warning) do not match. The Mimir Query Engine has AST optimisation where a right hand side expression will not be evaluated if the left side has not results
+* there are cases where query result annotations do not match on quantile percentile being out of bounds. The warning annotation includes a position in the original query of the invalid percentile and there is a difference in whitespace trimming of the query expressions between engines which results in a difference
+* there are some expressions not yet supported by the Mimir Query Engine, such as info, mad_over_time, sort_by_label and sort_by_label_desc
+* there are some expressions which do not have a deterministic result - ie topk(), bottomk() are undefined if there are multiple series with the same value
```

Comment:
Can we remove these from the test corpus, or somehow ensure that they run on expressions that have unique values?